src/widgets/SurveyAnswerProcessing/ui/ProcessingScreen.tsx

```diff
@@ -0,0 +1,76 @@
+import { useCallback, useContext } from 'react';
+
+import { ProgressBar } from './ProgressBar';
+
+import { appletModel } from '~/entities/applet';
+import { useBanners } from '~/entities/banner/model';
+import { SurveyContext, SurveyLayout, SurveyManageButtons } from '~/features/PassSurvey';
+import Box from '~/shared/ui/Box';
+import Text from '~/shared/ui/Text';
+import { useCustomTranslation } from '~/shared/utils';
+
+export const ProcessingScreen = () => {
+  const { t } = useCustomTranslation();
+
+  const context = useContext(SurveyContext);
+
+  const { addWarningBanner } = useBanners();
+
+  const { completeActivity, completeFlow } = appletModel.hooks.useEntityComplete({
+    activityId: context.activityId,
+    eventId: context.eventId,
+    appletId: context.appletId,
+    flow: context.flow,
+    flowId: context.flow?.id ?? null,
+    publicAppletKey: context.publicAppletKey,
+  });
+
+  const onFinish = useCallback(() => {
+    const canBeClosed = true; // TODO: Change on real one when the store will be ready
+
+    if (!canBeClosed) {
+      return addWarningBanner(t('answerProcessingScreen.processInProgress'));
+    }
+
+    return context.flow ? completeFlow({ forceComplete: true }) : completeActivity();
```

Comment:
How is it considered a "forced" complete action when `onFinish` is invoked by a user action (button press).

Reply:
The "forceComplete" flag is provided to the method to navigate the user after flow completion correctly.
BTW: This part of code very likely will be reimplemented in the next PR